docs/requirements.txt

```diff
@@ -1,3 +1,4 @@
 sphinx==5.0.1
 sphinxcontrib_trio==1.1.2
 sphinxcontrib-websupport
+typing_extensions>=4.2.0, <5
```

Comment:
Might not need this since it's a dependency of aiohttp?

Reply:
i dont think relying on another package's dependencies is a good idea personally

Reply:
The lib currently makes use of yarl in several places due to it being a dependency of aiohttp

Reply:
yes it would add safety just in case if
- aiohttp removes the dependency
- aiohttp bumps it above a major version which may break things